lib/src/main/kotlin/com/swmansion/starknet/deployer/Deployer.kt

```diff
@@ -0,0 +1,40 @@
+package com.swmansion.starknet.deployer
+
+import com.swmansion.starknet.data.types.Calldata
+import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.provider.Request
+import com.swmansion.starknet.provider.exceptions.RequestFailedException
+
+data class ContractDeployment(
+    val transactionHash: Felt,
+)
+
+class AddressRetrievalFailedException(message: String) : Exception(message)
```

Comment:
I think we can also include `ContractDeployment`. This way users will see in logs what exectly went wrong.